docs/content/preview/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/on-premises.md

```diff
@@ -750,6 +750,130 @@ As an alternative to setting crontab permissions, you can install systemd-specif
    WantedBy=timers.target
    ```
 
+### Use node agents
+
+To automate some of the steps outlined in [Provision nodes manually](#provision-nodes-manually), YugabyteDB Anywhere provides a node agent (also known as gRPC agent) that you can run on each node meeting the following requirements:
+
+- The node has already been set up with the `yugabyte` user group and home.
+- The bi-directional communication between the node and YugabyteDB Anywhere has been established (that is, the IP address can reach the host and vice versa). 
+
+### Installation
+
+You can install a node agent as follows:
+
+1. Download the installer from YugabyteDB Anywhere using the API token of the Super Admin, as follows:
+
+   ```sh
+   curl https://<yugabytedb_anywhere_address>/api/v1/node_agents/download --header 'X-AUTH-YW-API-TOKEN: <api_token>' > installer.sh && chmod +x installer.sh
+   ```
+
+3. Verify that the installer file contains the script.
+
+3. Run the following command to download the node agent's `.tgz` file which installs and starts the interactive configuration:
+
+   ```sh
+   ./installer.sh -t install -u https://<yugabytedb_anywhere_address> -at <api_token>
+   ```
+
+   For example, if you execute `./installer.sh  -t install -u http://100.98.0.42:9000 -at 301fc382-cf06-4a1b-b5ef-0c8c45273aef`, expect the following output:
+
+   ```output
+   * Starting YB Node Agent install
+   * Creating Node Agent Directory
+   * Changing directory to node agent
+   * Creating Sub Directories
+   * Downloading YB Node Agent build package
+   * Getting Linux/amd64 package
+   * Downloaded Version - 2.17.1.0-PRE_RELEASE
+   * Extracting the build package
+   * The current value of Node IP is not set; Enter new value or enter to skip: 10.9.198.2
+   * The current value of Node Name is not set; Enter new value or enter to skip: Test
+   * Select your Onprem Provider
+   1. Provider ID: 41ac964d-1db2-413e-a517-2a8d840ff5cd, Provider Name: onprem
+           Enter the option number: 1
+   * Select your Instance Type
+   1. Instance Code: c5.large
+           Enter the option number: 1
+   * Select your Region
+   1. Region ID: dc0298f6-21bf-4f90-b061-9c81ed30f79f, Region Code: us-west-2
+           Enter the option number: 1
+   * Select your Zone
+   1. Zone ID: 99c66b32-deb4-49be-85f9-c3ef3a6e04bc, Zone Name: us-west-2c
+           Enter the option number: 1
+           • Completed Node Agent Configuration
+           • Node Agent Registration Successful
+   You can install a systemd service on linux machines by running node-agent-installer.sh -t install-service (Requires sudo access).
+   ```
+
+4. Run the following command to enable the node agent as a systemd service, which is required for self-upgrade and other functions: 
+
+   ```sh
+   sudo node-agent-installer.sh -t install-service  
+   ```
+
```

Comment:
Once this installation is complete,  the configurations are saved in the file located at

`node-agent/config/config.yml`

The values should not be changed manually.

Reply:
Added